lychee-lib/src/utils/request.rs

```diff
@@ -53,14 +53,14 @@ fn try_parse_into_uri(raw_uri: &RawUri, source: &InputSource, base: &Option<Base
     let uri = match Uri::try_from(raw_uri.clone()) {
         Ok(uri) => uri,
         Err(_) => match base {
+            Some(Base::RootPath(_)) | None => match source {
+                InputSource::FsPath(root) => create_uri_from_file_path(root, &text, base)?,
+                _ => return Err(ErrorKind::UnsupportedUriType(text)),
+            },
             Some(base_url) => match base_url.join(&text) {
                 Some(url) => Uri { url },
                 None => return Err(ErrorKind::InvalidBaseJoin(text.clone())),
```

Comment:
I read this like
- If `--root-path` is set, try to append the root to the file path to create the full URL (this is the new behavior).
- Otherwise, if base_url is set, join the base URL with the given text to create the full URL (the existing behavior).

I'm not sure, but I *think* we need to focus on the `text` rather than the `Base` here. Because if `text` is a relative URL, we check if we have a base URL, whereas if `text` is an absolute URL, we check if we have a root path.
So, base URL and root path are orthogonal. They could be set at the same time and serve a different purpose.

Which raises the question if we should make `RootPath` a variant of `Base` or if it's an entirely separate type.

Sorry if that description is a bit vague, it's been a while since I looked into that part and the explanation might be wrong.